controllers/depresolver/depresolver_config.go

```diff
@@ -223,6 +278,44 @@ func parseMetricsAddr(metricsAddr string) (host string, port int, err error) {
 	return
 }
 
+func parseEdgeDNSServers(serverList string) []utils.DNSServer {
+	r := []utils.DNSServer{}
+	if len(strings.TrimSpace(serverList)) == 0 {
+		return r
+	}
+	chunks := strings.Split(serverList, ",")
+	var host, portStr string
+	var err error
+	for _, chunk := range chunks {
+		chunk = strings.TrimSpace(chunk)
+		switch strings.Count(chunk, ":") {
+		case 0: // ipv4 or domain w/o port
+			host = chunk
+			portStr = "53"
+		case 1: // ipv4 or domain w/ port
+			host, portStr, err = net.SplitHostPort(chunk)
+			if err != nil {
+				continue
+			}
+		default: // ipv6 or http://foo:bar
+			// not supported
+			continue
+		}
+		var port int
+		port, err = strconv.Atoi(portStr)
+		if err != nil {
+			port = 53
+		}
+		if host != "" {
+			r = append(r, utils.DNSServer{
+				Host: host,
+				Port: port,
+			})
+		}
+	}
+	return r
+}
```

Comment:
consider input `"127.0.0.1:5 053"` goes to valid `127.0.0.1:53`. 

Parsing servers is very complex operation in comparison with the rest, maybe we can retrieve errors from parser function and quit resolving immediatelly; or pass returning err into validation funcion which fails as well, but failing logic will be on the single place ( validation chunks in directly in a  parser may help a lot). 

 `if error_from_parsing != nil;  fail error message.`


Nit: ipv6 dismissed, do we need switch/case syntax now ?

Reply:
this is actually called only if the value of that env variable passes the reg exp check, which will not for `127.0.0.1:5 053`. I am adding a new test case w/ the `127.0.0.1:5 053` input so that you can see that it is not considered as a valid input (test called `TestResolveConfigWithInvalidEdgeDnsServerPort2`)

Reply:
perefect, there is large space of possible valid/invalid inputs, maybe, rather than create new tests,  you can iterate all invalid values you have on mind, like this. But decision is yours. 

```
for _, invalid := range []string{"127.0.0.1:5 053", "565:11", "host.cloud.example.com:5353:21".....} {
     // acting
     //. assertion
} 
```

Reply:
done